### DISCUSSION TITLE
Adding cache to the s3 adapter breaks the temporary file preview

### DISCUSSION THREAD

**Discussion by ricki-io** [livewire/livewire#3573] [7. Bugs (archived)]
To reproduce:

1. Fresh Laravel install
2. Use these library versions:
```
"league/flysystem-aws-s3-v3": "^1.0",
"league/flysystem-cached-adapter": "~1.0",
"livewire/livewire": "^2.5",
```

3. Set s3 temp file previews
`config(['livewire.temporary_file_upload.disk' => 's3'])`

4. Add cache to s3 adapter in `fillesystem.php`
```
's3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
            'endpoint' => env('AWS_ENDPOINT'),
            'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
            'cache' => [
                'store' => 'redis',
                'expire' => 600,
                'prefix' => 's3:',
            ]
        ],
```

5. Upload a file

You get:
`Call to undefined method League\Flysystem\Cached\CachedAdapter::getClient()` in `vendor/livewire/livewire/src/GenerateSignedUploadUrl.php:24`.

Not sure about this, but maybe Livewire should check the "instance of" `FileUploadConfiguration::storage()->getDriver()->getAdapter()` and if its a CachedAdapter use $adapter->s3Client and if its not then continue working like now.

Thanks.

<img width="349" alt="Screenshot 2021-08-24 at 19 06 10" src="https://user-images.githubusercontent.com/15916288/130601611-2d7cb246-8ca1-427e-b699-cb968bdd97a4.png">

<img width="363" alt="Screenshot 2021-08-24 at 19 06 02" src="https://user-images.githubusercontent.com/15916288/130601606-6038ac6b-76ac-4147-9813-078ad1adb2fd.png">

**Comment by joshhanley** [comment]
@ricard-pons ah yeah good catch! Would you be able to submit a PR with a fix for that?

Thanks!

**Reply by dutypro** [reply]
Any idea when this will be fixed? Still an issue seemingly.
